lib/MetaCPAN/Web/Controller/Activity.pm

```diff
@@ -6,75 +6,18 @@ BEGIN { extends 'MetaCPAN::Web::Controller' }
 
 use DateTime;
 
-my %res = ( week => '1w', month => 'month' );
-
 sub index : Path : Args(0) {
     my ( $self, $c ) = @_;
-    my $req = $c->req;
-    my $res = $res{ $req->parameters->{res} || 'week' } || '1w';
 
-    my $q = [];
-    if ( my $author = $req->parameters->{author} ) {
-        push( @$q, { term => { author => uc($author) } } );
-    }
-    if ( my $distribution = $req->parameters->{distribution} ) {
-        push( @$q, { term => { distribution => $distribution } } );
-    }
-    if ( my $module = $req->parameters->{module} ) {
-        push( @$q, { term => { 'dependency.module' => $module } } );
-    }
-    if ( $req->parameters->{f} && $req->parameters->{f} eq 'n' ) {
-        push( @$q,
-            { term  => { first  => 1 } },
-            { terms => { status => [qw< cpan latest >] } },
-        );
-    }
+    my %args = map { $_ => $c->req->parameters->{$_} }
```

Comment:
@oalders wasn't sure whether to create a new hash with the args or just pass `$c->req->parameters` to `request` - what do you think?